src_c/math.c

```diff
@@ -1633,18 +1633,52 @@ vector_reflect_ip(pgVector *self, PyObject *normal)
 static double
 _vector_distance_helper(pgVector *self, PyObject *other)
 {
-    Py_ssize_t i;
-    double distance_squared, tmp;
+    Py_ssize_t i, dim = self->dim;
+    double distance_squared = 0;
+
+    /* Specialised fastpath for Vector-Vector distance calculation*/
+    if (pgVector_Check(other)) {
+        pgVector *otherv = (pgVector *)other;
+        double dx, dy;
+
+        if (dim != otherv->dim) {
+            PyErr_SetString(PyExc_ValueError, "Vectors must be the same size");
+            return -1;
+        }
 
-    distance_squared = 0;
-    for (i = 0; i < self->dim; ++i) {
-        tmp = PySequence_GetItem_AsDouble(other, i) - self->coords[i];
-        distance_squared += tmp * tmp;
+        dx = otherv->coords[0] - self->coords[0];
+        dy = otherv->coords[1] - self->coords[1];
+
+        distance_squared = dx * dx + dy * dy;
+
+        if (dim == 3) {
```

Comment:
In the future, probably we should use dim loops for all code if it's doable. Because of a future Vector4/VectorN.

Reply:
> Or error out correctly if the code doesn't handle other dims.

Doesn't it already?

> In the future, probably we should use dim loops for all code if it's doable. Because of a future Vector4/VectorN.

Yeah this would have been better indeed